pkgs/development/haskell-modules/configuration-tensorflow.nix

```diff
@@ -4,61 +4,88 @@ with haskellLib;
+  proto-lens-setup = self.proto-lens-setup_0_4_0_2;
+  lens-family = self.lens-family_1_2_3;
 in
 {
-  proto-lens = appendPatch super.proto-lens ./patches/proto-lens-0.2.2.0.patch;
-  proto-lens-descriptors = doJailbreak (super.proto-lens-descriptors.override {
-    inherit proto-lens;
-    lens-labels = self.lens-labels_0_1_0_2;
+  lens-family_1_2_3 = super.lens-family_1_2_3.override {
+    lens-family-core = self.lens-family-core_1_2_3;
+  };
+  
+  proto-lens_0_5_1_0 = (appendPatch (doJailbreak super.proto-lens_0_5_1_0) ./patches/proto-lens-0.5.1.0.patch).override {
+    inherit lens-family;
+  };
+
+  proto-lens-runtime_0_5_0_0 = doJailbreak (super.proto-lens-runtime_0_5_0_0.override {
+    inherit lens-family proto-lens;
   });
-  proto-lens-protoc = appendPatch (addBuildDepend (super.proto-lens-protoc.override {
-    inherit proto-lens;
-  }) self.semigroups) ./patches/proto-lens-protoc-0.2.2.3.patch;
-  proto-lens-protobuf-types = doJailbreak (super.proto-lens-protobuf-types.override {
-    inherit proto-lens proto-lens-protoc;
+
+  proto-lens-protoc_0_5_0_0 = doJailbreak (super.proto-lens-protoc_0_5_0_0.override {
+    inherit lens-family proto-lens;
+    haskell-src-exts = self.haskell-src-exts_1_19_1;
   });
+  proto-lens-setup_0_4_0_2 = appendPatch (doJailbreak (super.proto-lens-setup_0_4_0_2.override {
+    inherit proto-lens-protoc;
+  })) ./patches/proto-lens-setup-0.4.0.2.patch;
 
-  lens-labels_0_1_0_2 = doJailbreak super.lens-labels_0_1_0_2;
+  proto-lens-protobuf-types_0_5_0_0 = doJailbreak (super.proto-lens-protobuf-types_0_5_0_0.override {
+    inherit lens-family proto-lens proto-lens-runtime proto-lens-setup;
+  });
 
-  haskell-src-exts_1_19_1 = appendPatch (doJailbreak super.haskell-src-exts_1_19_1) (
+  haskell-src-exts_1_19_1 = appendPatches (doJailbreak super.haskell-src-exts_1_19_1) [
     # Adapt to the Semigroup–Monoid Proposal (enables building on GHC >= 8.4)
-    pkgs.fetchpatch {
-      url = "https://github.com/haskell-suite/haskell-src-exts/commit/258e072fe9e37f94360b7488b58ea2832843bbb8.patch";
-      sha256 = "0ja6ai41v9plinlhjwja282m6ahn6mw4xi79np0jxqk83cg0z1ff";
-    }
-  );
+    (pkgs.fetchpatch {
+        url = https://github.com/haskell-suite/haskell-src-exts/commit/258e072fe9e37f94360b7488b58ea2832843bbb8.patch;
+        sha256 = "0ja6ai41v9plinlhjwja282m6ahn6mw4xi79np0jxqk83cg0z1ff";
+    })
+    # Adapt to MonadFail proposal (enables building on GHC >= 8.8)
+    (pkgs.fetchpatch {
+        url = https://gist.githubusercontent.com/mikesperber/0f2addaf3fbe97ffb4a5234d8711ba41/raw/e09e20998283c7195e82d546ba9266d290eb000d/gistfile1.txt;
+        sha256 = "18clvli7vpqsqlf2f3qidn71738j9zdlpana6wha3x0dxwan5ly0";
+    }) ];
 
-  tensorflow-proto = super.tensorflow-proto.override {
-    inherit proto-lens proto-lens-protobuf-types;
+  tensorflow-proto = (setSourceRoot "tensorflow-proto" (overrideCabal super.tensorflow-proto (drv: { src = tensorflow-haskell; }))).override {
```

Comment:
These lines might be a little simpler if you moved the `overrideCabal` part into the `setSourceRoot` function.

Reply:
Good idea. Done.